Develop

saihtaungkham opened the pull request:
First commit for adding the functionalities of MLSearch API requesters.

**Supported API Sources**
1. Paper with Code
2. Github

shripadtheneo committed:
Initial commit for python egg for library of api requester

Remove private access token

saihtaungkham committed:
Adding the functionalities to mlsearch package.

saihtaungkham committed:
Remove botocore as it seems to be removed by 2020/03.

saihtaungkham committed:
Remove old readme file.

rhezaharliman commented on src/main/python/mlsearch/helper.py:
```diff
@@ -0,0 +1,49 @@
+            return False
+    return True
+
+def response(message, status_code):
```
I found out that we still need to support CORS. I'll let you know the detail later in the meeting. 😄

saihtaungkham replied:
I fixed it at the lambda script. I guess it is an optional header and better to keep it out from ml_search api requester.

rhezaharliman commented on src/main/python/mlsearch/api_requester.py:
```diff
@@ -0,0 +1,161 @@
+
+        for item, typ in self.params_model.items():
+            if item in self.params.keys():
+                if not typ == type(self.params[item]):
```
I think we can make this into 1 condition only. How do you think?

saihtaungkham replied:
You are right. Thanks.

rhezaharliman commented on src/main/python/mlsearch/protocol.py:
```diff
@@ -0,0 +1,61 @@
+            if param not in param_list:
+                raise AttributeError('{} is not a valid parameter.'.format(param))
+        
+        self.title = kwargs.get('title', None)
```
it's s just my concern, CMIIW. I'm wondering what happened if there's no `title` and `None` is passed as the response. since it has a different type compared to an empty string or null.

saihtaungkham replied:
I confirmed it on json format and `None` is converted to `null`. I guess the client side need to confirm this behavior on their respective programming platform.

rhezaharliman reviewed (changes requested):
Thank you very much, Sai! I checked it briefly. I wrote some comments. I think Shripad will take a look at it deeper. 
Just CORS related that stuff that we might need to think about. Let's discuss it at the meeting. Thank you! 🙏

saihtaungkham reviewed (commented):
Thanks for your review. @rhezaharliman 🙏

saihtaungkham committed:
Fix rheza review comment.

rhezaharliman commented:
Thanks Sai. Looks good to me! May be Shripad wants to add something.

saihtaungkham committed:
Adding the POST request attributes.

saihtaungkham committed:
Removing the IP_address attribute.

saihtaungkham committed:
Updating the readme.

rhezaharliman reviewed (approved):
Thanks Sai! Looks good to me. 😄

saihtaungkham committed:
Adding and fixing the POST Attr missing.

rhezaharliman commented:
@saihtaungkham we might need to squash this commit https://github.com/Machine-Learning-Tokyo/search-api-requester/pull/1/commits/b4b65cfbbe4a070fe56e033c32c9330bfa8bbd3e 😅 

@shripadtheneo if you have free time, would you mind to take a look at it? thank you very much beforehand 🙏

saihtaungkham committed:
Adding the last_page flag.

saihtaungkham committed:
Fixing the response format.

saihtaungkham committed:
Adding Youtube API.

Modify Error Message

Update README.md

Update README.md

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/scripts/mlsearch

Update src/main/scripts/mlsearch

Update README.md

Update README.md

Update README.md

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update README.md

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/python/mlsearch/api_requester.py

Update src/main/scripts/mlsearch

Update src/main/python/mlsearch/helper.py

Update src/main/python/mlsearch/helper.py

Update src/main/scripts/mlsearch

Update src/main/scripts/mlsearch

Update src/main/scripts/mlsearch

Update src/main/scripts/mlsearch

Fix missing source in youtube.

saihtaungkham commented:
> b4b65cf
Oops... Private Access Token... 😅 
I will try to fix this.

saihtaungkham committed:
Fix html escape and code formatting.

saihtaungkham committed:
Fix int error on html unescape.

saihtaungkham committed:
Fix feedback bug and add features.

saihtaungkham committed:
Fix ambiguous youtube query order keyword.

saihtaungkham committed:
Add usage to README file.

saihtaungkham committed:
Fix YouTube response result and Exceptions.

saihtaungkham committed:
Check Auth Error on YouTube before happening.

saihtaungkham committed:
Improve Github Pagnition Logic.

saihtaungkham committed:
Fix YouTube response result and Exceptions.

Fix YouTube response result and Exceptions.

saihtaungkham committed:
Fix YouTube response result and Exceptions.

saihtaungkham committed:
Merge pull request #5 from Machine-Learning-Tokyo/fix/user_feedback

Fix/user feedback

saihtaungkham committed:
Fix variable name and code formatting.

saihtaungkham committed:
Merge pull request #6 from Machine-Learning-Tokyo/fix/github_pagination

Fix/GitHub pagination
